opt: minor cleanup around formatPrivate

RaduBerinde opened the pull request:
- move the `formatScanPrivate` code to `formatPrivate`

 - print out an `Ordering` private properly (relevant when formatting
   memo with Limit/Offset)

 - don't print out `SetOpColMap` private

Release note: None

RaduBerinde committed:
opt: minor cleanup around formatPrivate

 - move the `formatScanPrivate` code to `formatPrivate`

 - print out an `Ordering` private properly (relevant when formatting
   memo with Limit/Offset)

 - don't print out `SetOpColMap` private

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/25825)
<!-- Reviewable:end -->

rytaft commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but shouldn't there be more test output changes?

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD8sw-d3A3Oyb9cmqGC:b-c3qc2r)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!  The changes only affect the memo output; for expr output we only call `formatPrivate` for scans.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD8uYnP-tqwvbII7a5S:bbzkwea)*
<!-- Sent from Reviewable.io -->

andy-kimball commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD8w1eo4F0ClZk3dYLn:bnfp4nl)*
<!-- Sent from Reviewable.io -->

rytaft commented:
In that case, could you add a couple of test cases just to show that your `SetOpColMap` and `ExplainOpDef` changes make sense?

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD8w1Ai78b2xGsKfvBU:b-g6p3jq)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
The ExplainOpDef change is a no-op, like I said this code would only get called from the memo formatter (I added it just so the code is clearer).

There is already a testcase with memo with explain (xform/testdata/physprops/ordering) and there were no changes.

I added a test with union. The old output was `(union G2 G3 &{[%!s(opt.ColumnID=1) %!s(opt.ColumnID=2)] [%!s(opt.ColumnID=5) %!s(opt.ColumnID=6)] [%!s(opt.ColumnID=7) %!s(opt.ColumnID=8)]})`. I didn't need a test to know that not showing that would be an improvement :)

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD8xRyp77yma4_bdC8y:b-81wds2)*
<!-- Sent from Reviewable.io -->

rytaft commented:
:thumbsup: thanks for humoring me

---

Reviewed 2 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25825#-:-LD9-2pF1NdzhFVwtj-U:b-li3076)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
